src/pages/profile.tsx

```diff
@@ -78,7 +113,95 @@ export default function Page() {
       <Input background="white" 
         value={newCity} onChange={e => setNewCity(e.target.value)} />
     </FormControl>           
+
+    <Divider margin={componentSpacing} />
+
+    <InterviewPreference data={pref.interviews} 
+      updateData={updateInterviewPref} /> 
+
     <Button onClick={handleSubmit} variant="brand">保存</Button>
     {isLoading && <Loader loadingText='保存中...'/>}
   </VStack>;
 }
+
+function InterviewPreference({ data, updateData } : {
+  data: InterviewPref;
+  updateData: (data: InterviewPref) => void;
+}) {
+  const [noMoreThan, setNoMoreThan] = useState(data?.limit?.noMoreThan || 0);
+  const [until, setUntil] = useState(data?.limit?.until ? 
+    new Date(data.limit.until) : oneMonthDate);
+  const [optIn, setOptIn] = useState(data?.optIn ?? undefined);
+
+  useEffect(() => {
+    setNoMoreThan(data?.limit?.noMoreThan ?? 0);
+    setUntil(new Date(data?.limit?.until ?? oneMonthDate));
+    setOptIn(data?.optIn ?? undefined);
+  }, [data]);
+ 
+  const setLimit = (noMoreThan: number, until: Date) => {
+    const newData = structuredClone(data || {});
+    newData.limit = { noMoreThan, until: until.toISOString() };
+    updateData(newData);
+  };
+
+  const removeLimit = () => {
+    const newData = structuredClone(data);
+    delete newData?.limit;
+    updateData(newData);
+  };
+
+  const toggleOptIn = (optIn: boolean) => {
+    updateData({
+      ...data,
+      optIn: optIn ? true : undefined,
+    });
+  };
+
+  return <>
+    <FormControl>
+      <FormLabel>面试官偏好</FormLabel>
+      <Checkbox isChecked={optIn} onChange={e => toggleOptIn(e.target.checked)}>
+        我不是导师，但可以帮助面试学生。
+      </Checkbox>
+    </FormControl>
+
+    <FormControl>
+      <Flex wrap="wrap" alignItems="center">
+        <Checkbox isChecked={!!data?.limit}
+          onChange={e => e.target.checked ? setLimit(noMoreThan, until) : 
+            removeLimit()}>
+          面试限制：
+        </Checkbox>在
+        <DatePicker 
+          selected={until}
+          onChange={date => {
+            const newDate = date || oneMonthDate;
+            setUntil(newDate);
+            setLimit(noMoreThan, newDate);
+          }}
+          minDate={new Date()} maxDate={threeMonthsDate} disabled={!data?.limit}
+          customInput={<Input {...datePicker} textAlign="center" />}
+        />
+        之前，我还可以参与
+        <NumberInput 
+          value={noMoreThan} 
+          onChange={(_, value) => {
+            setNoMoreThan(value);
+            setLimit(value, until);
+          }}
+          min={0} max={10} isDisabled={!data?.limit}
+         >
+          <NumberInputField bgColor="white" />
```

Comment:
use theme styling for this one, too

Reply:
It will be covered by the other theme color setting.

Reply:
I'm not sure if I understand what you meant.

Reply:
I put it into the theme file and don't know why it is not working. Try both bgColor and bg not working.